src/daft-core/src/array/ops/get.rs

```diff
@@ -188,6 +190,29 @@ impl MapArray {
     }
 }
 
+impl FileArray {
+    #[inline]
+    pub fn get(&self, idx: usize) -> Option<DaftFile> {
+        let discriminant_array = self.discriminant_array();
+        let data_array = self.data_array();
+        let discriminant = discriminant_array.get(idx).unwrap();
```

Comment:
This can return None instead of panicing as the return type for the function is `Option`.